### PR TITLE
Fix running latex cell with cell marker

### LINKED ISSUE #10531: "SyntaxError: unexpected character after line continuation character" pops up
## Environment data

Build of Visual Studio Code - Insiders: 1.69.0-insider(user setup)
Build of python-Pre-Release: v2022.9.11681004
Build of jupyter-Pre-Release: v2022.6.1001741036
Build of pylance-Pre-Release: v2022.6.21
Build of jupyter Notebook Renderers: v1.0.8
Build of jupyter Keymap: v1.0.0

## Steps to reproduce:
1.Make sure Anaconda installed in local 
2.Open the file src/test/datascience/manualTestFiles/manualTestFile.py in VS Code 
3.At the top of the file it will list the things that you need installed in your Python environment 
4. On the first cell click Run Below 
5.Interactive Window should open, show connection information, and execute cells 
6.Check the "# %% Latex"

## Expected behaviour
Execute cells sucessfully without errors
image

## Actual behaviour
image

image

## Sample code

```
# %% Latex 

%%latex 

\begin{align} 

\nabla \cdot \vec{\mathbf{E}} & = 4 \pi \rho \\ 

\nabla \times \vec{\mathbf{E}}\, +\, \frac1c\, \frac{\partial\vec{\mathbf{B}}}{\partial t} & = \vec{\mathbf{0}} \\ 

\nabla \cdot \vec{\mathbf{B}} & = 0 

\end{align} 
```

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #10531

### CODE REVIEW

File: src/interactive-window/editor-integration/codeGenerator.ts [hunk 1]
```diff
@@ -76,6 +79,20 @@ export class CodeGenerator implements IInteractiveWindowCodeGenerator {
...
+        // If we do, then don't sen the cell marker, e.g. in the case of `latex` magic it cannot handle python style comments in the cell.
+        // I.e. %%latex magic cannot handle lines that start with the text `#` which exists in cell markers such as `# %%`.
+        if (
+            metadata.interactiveWindowCellMarker &&
+            lines.length &&
+            lines[0].trim() === metadata.interactiveWindowCellMarker.trim()
+        ) {
+            const nonEmptyLines = lines.slice(1).filter((line) => line.trim().length > 0);
+            if (nonEmptyLines.length > 0 && nonEmptyLines[0].trim().startsWith('%%')) {
+                return { lines: lines.slice(1), executableLines: lines.slice(1) };
```
Comment (rchiodo) [inline src/interactive-window/editor-integration/codeGenerator.ts:92]: Shouldn't this be nonEmptyLines? Otherwise how do you know to slice the first line?

Comment (DonJayamanne) [reply src/interactive-window/editor-integration/codeGenerator.ts:92]: I didn't want to strip out the white space, I guess I can do that as well.

**Review by rchiodo — APPROVED** [review]

File: src/interactive-window/editor-integration/codeGenerator.ts [hunk 2]
```diff
@@ -76,6 +79,21 @@ export class CodeGenerator implements IInteractiveWindowCodeGenerator {
             lines.forEach((line, index) => (lines[index] = uncommentMagicCommands(line)));
         }
 
+        // If we have a cell magic then don't send the cell marker.
+        // If we do, then don't sen the cell marker, e.g. in the case of `latex` magic it cannot handle python style comments in the cell.
```
Comment (amunger) [inline src/interactive-window/editor-integration/codeGenerator.ts:83]: repetitive comment with the previous line

File: src/test/datascience/editor-integration/codeGenerator.unit.test.ts [hunk 3]
```diff
@@ -84,6 +84,51 @@ suite('Code Generator Unit Tests', () => {
         assert.equal(generatedCodes[0].generatedCodes[0].executionCount, 1, 'Wrong execution count');
     });
 
+    test('Execute %%latex magic in a cell with a cell marker', async () => {
```
Comment (amunger) [inline src/test/datascience/editor-integration/codeGenerator.unit.test.ts:87]: ha, I was about to mention that this change was a good candidate for unit tests

**Review by amunger — APPROVED** [review]

File: src/test/datascience/editor-integration/codeGenerator.unit.test.ts (hunk 3 above)
Comment (DonJayamanne) [reply src/test/datascience/editor-integration/codeGenerator.unit.test.ts:87]: Had to add an integration test as this was a bad break, basically as bad as not being able to run a cell.
To me this is a golden scenario, not being able to run latext cells in IW.

**Pull request merged**